src/main/scala/is/hail/rvd/OrderedRVDType.scala

```diff
@@ -43,6 +45,8 @@ class OrderedRVDType(
   val pkInKOrd: UnsafeOrdering = OrderedRVDType.selectUnsafeOrdering(kType, pkKFieldIdx, kType, pkKFieldIdx)
   val kRowOrd: UnsafeOrdering = OrderedRVDType.selectUnsafeOrdering(kType, (0 until kType.size).toArray, rowType, kRowFieldIdx)
 
+  val pkRowEOrd: ExtendedOrdering = ExtendedOrdering.extendToNull(pkRowOrd)
```

Comment:
This can't be right.  pkRowOrd is a RegionValue ordering, but ExtendedOrderings are only allowed for orderings on annotations.  unsafe orderings handle missing by default.  I think you probably just want `pkType.ordering`.  Also, it's a val, so you don't really need to make it a variable here.

Reply:
Yes, I checked, both uses should just use pkType.ordering (which is an extended ordering).

Reply:
should I also extract the pk from the row before comparing here, too?

Reply:
I think you should remove pkRowEOrd and call pkType.ordering where you're using it.

Reply:
that involves extracting the pk from the row I'm comparing to, right? I'm not doing that right now because pkRowOrd compares a RegionValue[pkType] to the key fields in RegionValue[rowType].

Reply:
Correct.

Right, and you can't use pkRowOrd here because it operates on region value but you're calling all of this on annotations, not region values.